deepspeed/runtime/config.py

```diff
@@ -521,8 +521,10 @@ def write_config(self, filename):
 class DeepSpeedConfig(object):
     def __init__(self, json_file, mpu=None, param_dict=None):
         super(DeepSpeedConfig, self).__init__()
-
-        if param_dict is None:
+        if isinstance(json_file, dict):

```

Comment:
I think typing json_file as a dict is confusing, rather we should pass dict as param_dict. So ideally this file should be unchanged for this PR. Instead the following should be modified to optionally pass param_dict as appropriate. 
[1](https://github.com/microsoft/DeepSpeed/blob/4b31cc2ed994feeff1a5def1cd768f65a5a7b349/deepspeed/runtime/zero/partition_parameters.py#L395)
[2](https://github.com/microsoft/DeepSpeed/blob/4b31cc2ed994feeff1a5def1cd768f65a5a7b349/deepspeed/runtime/zero/partition_parameters.py#L411)